drivebrain_core_impl/drivebrain_comms/include/DBServiceImpl.hpp

```diff
@@ -15,14 +18,17 @@ class DBInterfaceImpl final : public db_service::v1::service::DBInterface::Servi
     };
     grpc::Status RequestStopLogging(grpc::ServerContext* context, const google::protobuf::Empty *rq, db_service::v1::service::LoggerStatus * response) override; 
     grpc::Status RequestStartLogging(grpc::ServerContext* context, const google::protobuf::Empty *rq, db_service::v1::service::LoggerStatus * response) override;
-    grpc::Status RequestCurrentLoggerStatus(grpc::ServerContext* context, const google::protobuf::Empty* request, db_service::v1::service::LoggerStatus* response) override; 
+    grpc::Status RequestCurrentLoggerStatus(grpc::ServerContext* context, const google::protobuf::Empty* rq, db_service::v1::service::LoggerStatus* response) override;
+    grpc::Status RequestControllerChange(grpc::ServerContext* context, const db_service::v1::service::DesiredController* rq, db_service::v1::service::ControllerChangeStatus* response) override; 
     
     public: 
-        DBInterfaceImpl(std::shared_ptr<core::MsgLogger<std::shared_ptr<google::protobuf::Message>>> logger_inst);
+        DBInterfaceImpl(std::shared_ptr<core::MsgLogger<std::shared_ptr<google::protobuf::Message>>> logger_inst, std::shared_ptr<control::ControllerManager<control::Controller<core::ControllerOutput, core::VehicleState>, 2>> ctr_manager_inst, core::StateEstimator* state_estimator);
         void run_server(); 
         void stop_server();
     private:
         std::shared_ptr<core::MsgLogger<std::shared_ptr<google::protobuf::Message>>> _logger_inst;
+        std::shared_ptr<control::ControllerManager<control::Controller<core::ControllerOutput, core::VehicleState>, 2>> _ctr_manager_inst;
```

Comment:
https://github.com/hytech-racing/drivebrain_software/blob/master/drivebrain_app/main.cpp#L114 this is how to register the member functions

implementation of msglogger:

https://github.com/hytech-racing/drivebrain_software/blob/master/drivebrain_core_base/include/public/MsgLogger.hpp#L30

Reply:
switching to a vector makes sense to not hard code controller number, but if we pass in funcs instead of the controllers, would there not need to be multiple vectors for each individual function that the controllers need to be called on, right now I would need one vector for the get dt secs and one for the stepping(meaning if more functions are added to the controller then I would need more and more vectors), does it not make more sense to have a vector of the controllers themselves? I could also be missing something about what u mean by the function, but the msglogger seems to be using much more singular, standalone methods

Reply:
> switching to a vector makes sense to not hard code controller number, but if we pass in funcs instead of the controllers, would there not need to be multiple vectors for each individual function that the controllers need to be called on, right now I would need one vector for the get dt secs and one for the stepping(meaning if more functions are added to the controller then I would need more and more vectors), does it not make more sense to have a vector of the controllers themselves? I could also be missing something about what u mean by the function, but the msglogger seems to be using much more singular, standalone methods

I would prefer that we did not use a vector of controller numbers. we do not change the number of controllers at runtime which would be a reason to use a vector.

A template argument is not a "hard coded" value when it comes to this.

also when it comes to the passing of funcs i am talking about using them to pass in the `swap_active_controller` member function from the controller manager. im not quite sure what you are referring to with the multiple vectors for each individual function.

Reply:
misunderstood this to mean remove the template and pass in funcs to the ctr manager, but alr did that, just got it fully unpointered